access/validator.go

```diff
@@ -231,6 +237,37 @@ func (v *TransactionValidator) checkAddresses(tx *flow.TransactionBody) error {
 	return nil
 }
 
+func (v *TransactionValidator) checkSignatureFormat(tx *flow.TransactionBody) error {
+
+	for _, signature := range append(tx.PayloadSignatures, tx.EnvelopeSignatures...) {
+		// check the format of the signature is valid.
+		// a valid signature is an ECDSA signature of either P-256 or secp256k1 curve.
+		ecdsaSignature := signature.Signature
+
+		// check if the signature could be a P-256 signature
+		valid, err := crypto.SignatureFormatCheck(crypto.ECDSAP256, ecdsaSignature)
+		if err != nil {
+			return fmt.Errorf("could not check the signature format (%s): %w", signature, err)
+		}
+		if valid {
+			continue
+		}
+
+		// check if the signature could be a secp256k1 signature
+		valid, err = crypto.SignatureFormatCheck(crypto.ECDSASecp256k1, ecdsaSignature)
+		if err != nil {
+			return fmt.Errorf("could not check the signature format (%s): %w", signature, err)
+		}
+		if valid {
+			continue
+		}
+
+		return InvalidSignatureError{Signature: signature}
```

Comment:
```suggestion
		return InvalidSignatureError{Signature: signature, Msg: "not a ECDSAP256 signature nor a ECDSASecp256k1 signature"}
```

Better UX if we could explain why it is considered as invalid.